src/OrchardCore.Modules/OrchardCore.Tenants/Services/TenantValidator.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Localization;
+using OrchardCore.Data;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Mvc.ModelBinding;
+using OrchardCore.Tenants.ViewModels;
+
+namespace OrchardCore.Tenants.Services
+{
+    public class TenantValidator
+    {
+        private readonly IShellHost _shellHost;
+        private readonly IFeatureProfilesService _featureProfilesService;
+        private readonly IEnumerable<DatabaseProvider> _databaseProviders;
+        private readonly ShellSettings _shellSettings;
+        private readonly IStringLocalizer<TenantValidator> S;
+
+        public TenantValidator(
```

Comment:
This doesn't fix the actual bug when you try from the admin: https://github.com/OrchardCMS/OrchardCore/issues/10269 Furthermore, now there's no validation error when you create two tenants with the exact same host as well. Please add this to the test too.

Reply:
Let me doble check, seems something weired happen after the latest changes